astropy/time/core.py

```diff
@@ -531,6 +531,83 @@ def value(self):
         """Time value(s) in current format"""
         return self._shaped_like_input(self._time.to_value(parent=self))
 
+    def light_travel_time(self, skycoord, kind='barycentric', location=None):
+        """Light travel time correction to the barycentre or heliocentre.
+
+        The frame transformations used to calculate the location of the solar
+        system barycentre and the heliocentre rely on the erfa routine epv00,
+        which is consistent with the JPL DE405 ephemeris to an accuracy of
+        11.2 km, corresponding to a light travel time of 4 microseconds.
+
+        The routine assumes the source(s) are at large distance, i.e., neglects
+        finite-distance effects.
+
+        Parameters
+        ----------
+        skycoord: `~astropy.coordinates.SkyCoord`
+            The sky location to calculate the correction for.
+        kind: str, optional
+            ``'barycentric'`` (default) or ``'heliocentric'``
+        location: `~astropy.coordinates.EarthLocation`, optional
+            The location of the observatory to calculate the correction for.
+            If no location is given, the ``location`` attribute of the Time
+            object is used
+
+        Returns
+        -------
+        time_offset: `~astropy.time.TimeDelta`
+            The time offset between the barycentre or Heliocentre and Earth,
+            in TDB seconds.  Should be added to the original time to get the
+            time in the Solar system barycentre or the Heliocentre.
+        """
+
+        if kind.lower() not in ('barycentric', 'heliocentric'):
+            raise ValueError("'kind' parameter must be one of 'heliocentric' "
+                             "or 'barycentric'")
+
+        if location is None:
+            if self.location is None:
+                raise ValueError('An EarthLocation needs to be set or passed '
+                                 'in to calculate bary- or heliocentric '
+                                 'corrections')
+            location = self.location
+
+        from ..coordinates import (UnitSphericalRepresentation, CartesianRepresentation,
+                                   HCRS, ICRS, GCRS, EarthLocation, SkyCoord)
+
+        if not isinstance(location, EarthLocation):
```

Comment:
I feel these types of tests are not needed -- someone might want to duck-type with a class that has a `get_itrs` method, and why not let them? -- but if you keep it, you might as well put is as an `elif not ...` above for the case that `location is not None` since the internal one should be validated already.